leptos/src/suspense_component.rs

```diff
@@ -61,14 +61,14 @@ pub fn Suspense<F, E, V>(
     /// Returns a fallback UI that will be shown while `async` [`Resource`](leptos_reactive::Resource)s are still loading.
     fallback: F,
     /// Children will be displayed once all `async` [`Resource`](leptos_reactive::Resource)s have resolved.
-    children: Box<dyn Fn() -> V>,
+    children: Rc<dyn Fn() -> V>,
 ) -> impl IntoView
 where
     F: Fn() -> E + 'static,
     E: IntoView,
     V: IntoView + 'static,
 {
-    let orig_children = Rc::new(children);
```

Comment:
Nice! Clearly I'd already encountered the same issue and just... added another `Rc<_>` instead of dealing with it :smile:

Reply:
Yeah this is clearly the kind of situation this PR is trying to solve, I just came across this one but there is maybe other built in components that could take advantages of this. I might look into it